recipes/librhash/all/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+cmake_minimum_required(VERSION 3.1.2)
+project(librhash LANGUAGES C VERSION ${LIBRHASH_VERSION})
+include(GNUInstallDirs)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(MSVC OR MSVC90 OR MSVC10)
+    set(MSVC ON)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
+endif()
+
+option(LIBRHASH_OPENSSL "enable OpenSSL (optimized hash functions) support")
+option(LIBRHASH_OPENSSL_RUNTIME "Load OpenSSL at runtime if present")
+
+if(LIBRHASH_OPENSSL)
+    find_package(OpenSSL REQUIRED)
+endif()
+
+set(SOURCE_SUBFOLDER ${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder)
+set(LIBRHASH_LIBRARY librhash)
+set(LIBRHASH_PROGRAM librhash)
+set(LIBRHASH_CONFIG LibRHashConfig)
+set(LIBRHASH_CMAKE_NAMESPACE LibRHash::)
+
+add_library(${LIBRHASH_LIBRARY}
+    "${SOURCE_SUBFOLDER}/calc_sums.c"
```

Comment:
I'm not confident using this part for a generic recipe, it may change on a next version. You can try something similar what vcpkg did: https://github.com/microsoft/vcpkg/blob/master/ports/rhash/CMakeLists.txt They extract the source file list from the Makefile

Reply:
I think vcpkg is wrong here because it is extracting the `SOURCES` variable where I think it should extract `LIBRHASH_FILES`.
The files in the `SOURCES` variable are for the `rhash` executable.
I changed it into something similar but I do the parsing in the `conanfile.py` recipe because the regex would become too complicated (files are spread over multiple lines).

I checked rhash/RHash#71, where they also make the distinction between files in the root folder and files in the librhash subfolder.

Reply:
At the moment, I am passing the sources as an argument to the cmake file during configuration.
Would this be fine? Generating a `CMakeLists.txt` during `source()` that contains the sources and add that file to the recipe. Or would you generate this cmake file during the build step? Or is the current approach fine?